src/WebJobs.Extensions.DurableTask/LocalGrpcListener.cs

```diff
@@ -346,10 +313,12 @@ private DurabilityProvider GetDurabilityProvider(ServerCallContext context)
                 return this.extension.GetDurabilityProvider(attribute);
             }
 
-            private OrchestrationStatus[] GetStatusesNotToOverride()
+            private IDurableClient GetClient(ServerCallContext context)
             {
-                OverridableStates overridableStates = this.extension.Options.OverridableExistingInstanceStates;
-                return overridableStates.ToDedupeStatuses();
+                string? taskHub = context.RequestHeaders.GetValue("Durable-TaskHub");
+                string? connectionName = context.RequestHeaders.GetValue("Durable-ConnectionName");
+                var attribute = new DurableClientAttribute() { TaskHub = taskHub, ConnectionName = connectionName };
+                return this.extension.GetClient(attribute);
```

Comment:
The above three lines are equivalent to those in `GetDurabilityProvider`. Should we factor them out into a separate method to make the code more robust to potential future changes in the gRPC header contract?